### DISCUSSION TITLE
Typescript doesn't include the foreign key types

### DISCUSSION THREAD

**Discussion by beshur** [prisma/prisma#21318] [Q&A] [answered]
### Question

Hi.

For some reason the generated client does not contain typings for model foreign keys until I explicitly define them like this:
`invoice: Invoices & {Owners: Owners} & {Packages: Packages[]}`

Otherwise I get error: 
`Property 'Owners' does not exist on type 'GetResult<{ id: number; total: number; currency: string; status: Invoices_status; ownerId: number; id_Owners: number; createdAt: Date; updatedAt: Date; PromoCodesId: number; }, unknown> & {}'.ts(2339)`

schema.prisma:
```
model Invoices {
  id           Int             @id @default(autoincrement())
  total        Int             @default(0)
  currency     String          @default("EUR") @db.Char(3)
  status       Invoices_status @default(pending)
  ownerId      Int
  id_Owners    Int?
  createdAt    DateTime        @default(now()) @db.DateTime(0)
  updatedAt    DateTime        @default(now()) @db.DateTime(0)
  PromoCodesId Int?
  PromoCodes   PromoCodes?     @relation(fields: [PromoCodesId], references: [id])
  Owners       Owners?         @relation(fields: [id_Owners], references: [id], onDelete: Restrict, onUpdate: Restrict, map: "Invoices_ibfk_1")
  Payments     Payments[]
  Packages     Packages[]      @relation("InvoicesToPackages")

  @@index([id_Owners], map: "id_Owners")
  @@index([PromoCodesId], map: "Invoices_PromoCodesId_fkey")
}
```

**Comment by nurul3101** [comment, marked as answer]
Hello @beshur 👋 

This is an expected behaviour. By default, only the scalar fields are included in the generated types.

If you want to include the relation fields, you need to use `Prisma.validator` function to create those types.

Here's a reference on how you can do it: [Using variations of generated model types](https://www.prisma.io/docs/concepts/components/prisma-client/advanced-type-safety/operating-against-partial-structures-of-model-types#problem-using-variations-of-the-generated-model-type).

It would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 
Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by nurul3101** [reply]
@beshur Did you had a chance to look at my last response? Please let us know in case you still need any help. If not, please consider marking the discussion as answered/closed.

**Reply by beshur** [reply]
Thanks, yes, I have seen it, but did not have time to verify it.

**Reply by nurul3101** [reply]
Thanks for getting back. Let us know in case you run into any issues while trying it out.

**Reply by beshur** [reply]
Well, I finally got to it :)
Worked like a charm, thank you!
